core/src/test/java/org/infinispan/util/BaseControlledConsistentHashFactory.java

```diff
@@ -19,27 +22,38 @@
  * @since 6.0
  */
 @SuppressWarnings("unchecked")
-public abstract class BaseControlledConsistentHashFactory implements ConsistentHashFactory<DefaultConsistentHash>,
+public abstract class BaseControlledConsistentHashFactory<CH extends ConsistentHash> implements ConsistentHashFactory<CH>,
```

Comment:
I could create two distinct base classes for dist/scattered cache, and then again in extending classes... this way has the least duplicated code through sort of Strategy pattern.
As for the name, I'd agree, but I won't rename existing classes unless the class name is unfit - and Base is understandable.